index.js

```diff
@@ -3,29 +3,28 @@ var pushable = require('pull-pushable')
 
 module.exports = function () {
   var listeners = []
+  var closed = false
 
   function notify (message) {
     // notify by pushing to all listeners
     for (var i = 0; i < listeners.length; i++) {
+      if (closed) return message
       listeners[i].push(message)
     }
     return message
   }
 
   notify.listen = function () {
     // create listener with `onClose` handler
-    var listener = pushable(function onClose () {
-      // if listener is found, delete from list
-      var index = listeners.indexOf(listener)
-      if (index !== -1) listeners.splice(index, 1)
-    })
+    var listener = pushable()
     listeners.push(listener)
     return listener
   }
 
   notify.abort = function (err) {
     // abort by ending all listeners
-    while (listeners.length) listeners[0].end(err)
+    closed = true
+    while (listeners.length) listeners.shift().end(err)
```

Comment:
if there are a lot of listeners, this can be quite slow as you create a new array on each shift. I think it's a minor thing, just wanted to note it here.

Reply:
Hmm, I believe `shift()` just mutates the current array, doesn't clone the existing array. At least that's what it seems from https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/shift

Or are you talking about something else?

Reply:
I'll merge and release because I want to get other dependencies using this, but we can revise this if you want :)

Reply:
Yeah, as said I don't think it makes a big difference in real life. To be concrete: https://gist.github.com/arj03/4eee4b4950916e2cd1b011f600964d3a is a test for 10k elements. `clear` is roughly 4 times faster than `shift`. Increase that number to 100k and you get 365x faster. I don't see how anyone would have that many listeners, but I just wanted to know the difference in general because this pattern does come up once in a while.